contracts/integrations/nevermined/AgentFactorySubscription.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.21;
+
+import {AgentMechSubscription} from "./AgentMechSubscription.sol";
+import {GenericManager} from "../../../lib/autonolas-registries/contracts/GenericManager.sol";
+
+interface IAgentRegistry {
+    /// @dev Creates a agent.
+    /// @param agentOwner Owner of the agent.
+    /// @param agentHash IPFS CID hash of the agent metadata.
+    /// @return agentId The id of a minted agent.
+    function create(address agentOwner, bytes32 agentHash) external returns (uint256 agentId);
+}
+
+/// @title Agent Factory Subscription - Periphery smart contract for managing agent and mech creation with subscription
+contract AgentFactorySubscription is GenericManager {
+    event CreateMech(
+        address indexed mech,
+        uint256 indexed agentId,
+        uint256 creditsPerRequest,
+        address indexed subscriptionNFT,
+        uint256 subscriptionTokenId
+    );
+
+    // Agent factory version number
+    string public constant VERSION = "1.0.0";
+
+    // Agent registry address
+    address public immutable agentRegistry;
+
+    constructor(address _agentRegistry) {
+        agentRegistry = _agentRegistry;
+        owner = msg.sender;
+    }
+
+    /// @dev Creates agent.
+    /// @param agentOwner Owner of the agent.
+    /// @param agentHash IPFS CID hash of the agent metadata.
+    /// @param creditsPerRequest Number of credits to pay for request via subscription.
+    /// @param subscriptionNFT Subscription address.
+    /// @param subscriptionTokenId Subscription token Id.
+    /// @return agentId The id of a created agent.
+    /// @return mech The created mech instance address.
+    function create(
+        address agentOwner,
+        bytes32 agentHash,
+        uint256 creditsPerRequest,
```

Comment:
Can these ever be updated? The idea was to have dynamic pricing per request...

Reply:
The price can be updated by the Mech's `setPrice()` function